v3/src/components/slider/slider-component-handler.ts

```diff
@@ -0,0 +1,202 @@
+    scaleType,
+    dateMultipleOfUnit,
+  } = values as V2Slider
+
+  const snapshot: Partial<ISliderSnapshot> = {}
+
+  if (globalValueName) {
+    const { document } = appState
+    const globalManager = document.content?.getFirstSharedModelByType(GlobalValueManager)
+    const global = globalManager?.getValueByName(globalValueName)
+    if (!global) {
+      return {error: errorResult(t("V3.DI.Error.globalNotFound", { vars: [globalValueName] }))}
+    }
+    snapshot.globalValue = global.id
+
+    // If this is a new slider, or the global is being changed, check for multiple sliders
+    if (!sliderModel || sliderModel.globalValue.id !== global.id) {
+      // Multiple sliders for one global value are not allowed
+      let existingTile = false
+      document.content?.tileMap.forEach(sliderTile => {
+        if (isSliderModel(sliderTile.content) && sliderTile.content.globalValue.id === global.id) {
+          existingTile = true
+        }
+      })
+      if (existingTile) {
+        return {error: errorResult(t("V3.DI.Error.noMultipleSliders", { vars: [globalValueName] }))}
+      }
+    }
+  }
+
+  if (dateMultipleOfUnit != null) {
+    if (!dateUnits.includes(dateMultipleOfUnit as any)) {
+      return {error: errorResult(t("V3.DI.Error.unsupportedDateUnit", { vars: [dateMultipleOfUnit] }))}
+    }
+    snapshot.dateMultipleOfUnit = dateMultipleOfUnit as ISliderSnapshot["dateMultipleOfUnit"]
+  }
+
+  if (scaleType != null) {
+    if (!SliderScaleTypes.includes(scaleType as any)) {
+      return {error: errorResult(t("V3.DI.Error.unsupportedScaleType", { vars: [scaleType] }))}
+    }
+    snapshot.scaleType = scaleType as ISliderSnapshot["scaleType"]
+  }
+
+  if (_animationDirection != null) {
+    const directionNumber = Number(_animationDirection)
+    if (isNaN(directionNumber) || directionNumber < 0 || directionNumber >= AnimationDirections.length) {
+      return {error: errorResult(t("V3.DI.Error.unsupportedAnimationDirection", { vars: [_animationDirection] }))}
+    }
+    snapshot.animationDirection = AnimationDirections[directionNumber]
+  }
+
+  if (_animationMode != null) {
+    const modeNumber = Number(_animationMode)
+    if (isNaN(modeNumber) || modeNumber < 0 || modeNumber >= AnimationModes.length) {
+      return {error: errorResult(t("V3.DI.Error.unsupportedAnimationMode", { vars: [_animationMode] }))}
+    }
+    snapshot.animationMode = AnimationModes[modeNumber]
+  }
+
+  return { validatedValues: snapshot }
+}
+
+export const sliderComponentHandler: DIComponentHandler = {
+  create({ values }) {
+    const {
+      animationRate,
+      multipleOf,
+      lowerBound,
+      upperBound,
+    } = values as V2Slider
+    const { error, validatedValues } = validateInput(values)
+    if (error) {
+      return error
+    }
+    const {
+      animationDirection,
+      animationMode,
+      dateMultipleOfUnit,
+      globalValue,
+      scaleType,
+    } = validatedValues as Partial<ISliderSnapshot>
```

Comment:
I think the only reason to cast `validatedValues` here is if it's `undefined`. So I think it would be better to throw an error if `validateValues` doesn't exist (which will probably never happen if there's no error but who knows what the future will bring). Then you wouldn't have to cast here. I think this will cause an error if `validatedValues` is ever `undefined`.